syntax = "proto3";

package org.couchers.api.donations;

import "annotations.proto";

service Donations {
  option (auth_level) = AUTH_LEVEL_SECURE;

  rpc InitiateDonation(InitiateDonationReq) returns (InitiateDonationRes) {
    // Initiate a donation
  }
}

message InitiateDonationReq {
  // amount in USD, integral for now
  uint32 amount = 1;
  // whether to create a monthly recurring donation
  bool recurring = 2;
}

message InitiateDonationRes {
  string stripe_checkout_session_id = 1;
}
